lib/models/application-invite.ts

```diff
@@ -0,0 +1,280 @@
+	 * });
+	 *
+	 * @example
+	 * balena.models.applicationInvite.getAllByApplication(applicationId, function(error, applicationInvites) {
+	 * 	if (error) throw error;
+	 * 	console.log(applicationInvites);
+	 * });
+	 */
+	exports.getAllByApplication = function(
+		applicationId: Application['id'],
+		options: PineOptionsFor<ApplicationInvite> = {},
+		callback?: (error?: Error, result?: ApplicationInvite[]) => void,
+	): Promise<ApplicationInvite[]> {
+		callback = findCallback(arguments);
+
+		return pine
+			.get({
+				resource: RESOURCE,
+				options: mergePineOptions(
+					{
+						$filter: {
+							is_invited_to__application: applicationId,
+						},
+						$expand: {
+							invitee: {
+								$select: ['id', 'email'],
+							},
+							application_membership_role: {
+								$select: ['id', 'name'],
+							},
+						},
+						$select: ['id', 'message', 'created_at'],
+					},
+					options,
+				),
+			})
+			.asCallback(callback);
+	};
+
+	/**
+	 * @summary Creates a new invite for an application
+	 * @name create
+	 * @public
+	 * @function
+	 * @memberof balena.models.applicationInvite
+	 *
+	 * @description This method invites a user by thier email/username to an application.
+	 *
+	 * @param {number} applicationId - the application id
+	 * @param {String} collaborator - the email/username of invitee
+	 * @param {String} roleTitle - the role title to be granted to the invitee
+	 * @param {String} [message=null] - the message to send along with the invite
+	 *
+	 * @fulfil {String} - application invite id
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.applicationInvite.create(applicationId,"invitee@example.org", "developer","join my app").then(function(applicationInvite) {
+	 * 	console.log(applicationInvite);
+	 * });
+	 *
+	 * @example
+	 * balena.models.applicationInvite.create(applicationId,g_invitee, "operator").then(function(applicationInvite) {
+	 * 	console.log(applicationInvite);
+	 * });
+	 *
+	 * @example
+	 * balena.models.applicationInvite.create(applicationId, g_invitee, "operator", function(error, applicationInvite) {
+	 * 	if (error) throw error;
+	 * 	console.log(applicationInvite);
+	 * });
+	 */
+	exports.create = function(
+		applicationId: Application['id'],
+		collaborator: string,
+		roleTitle: ApplicationMembershipRole['name'],
+		message?: ApplicationInvite['message'],
+	) {
+		return pine
+			.get<ApplicationMembershipRole>({
+				resource: 'application_membership_role',
+				options: {
+					$select: ['id'],
+					$filter: {
+						name: roleTitle,
+					},
+				},
+			})
+			.get(0)
+			.then((role: ApplicationMembershipRole) => {
+				return role.id;
+			})
+			.then(roleId => {
+				return pine.post({
+					resource: RESOURCE,
+					body: {
+						collaborator,
+						is_invited_to__application: applicationId,
+						message,
+						application_membership_role: roleId,
+					},
+				});
+			});
+	};
+
+	/**
+	 * @summary Revoke an Application Invite
```

Comment:
Again the casing is inconsistent with the rest of the messages, let's stick with one format.